### PR TITLE
escape `#` in encoder so `#{` can't happen in resulting string

### PR DESCRIPTION
**Pull request by repomaa**

### CODE REVIEW

File: spec/storage/string_interpolation/test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+L#9U"gAc-XLA*/Ac|1M:
```
Comment (straight-shoota) [inline spec/storage/string_interpolation/test.txt]: Could you please add a description for this file's purpose that it doesn't get lost?

File: src/loader/loader.cr [hunk 2]
```diff
@@ -69,7 +69,7 @@ module BakedFileSystem
       def write(slice : Bytes)
         slice.each do |byte|
           case byte
-          when 35_u8..91_u8, 93_u8..127_u8
+          when 36_u8..91_u8, 93_u8..127_u8
```
Comment (straight-shoota) [inline src/loader/loader.cr]: Maybe add `when 35_u8; @io << "\\#"`? That would result in a few bytes less in the baked string.

Comment (straight-shoota) [inline src/loader/loader.cr:72]: Maybe add `when 35_u8; @io << "\\#"`? That would result in a few bytes less in the baked string.

File: spec/storage/string_interpolation/test.txt (hunk 1 above)
Comment (straight-shoota) [inline spec/storage/string_interpolation/test.txt:1]: Could you please add a description for this file's purpose so it doesn't get lost?

**Review by straight-shoota — CHANGES_REQUESTED** [review]
Good catch! Seems like I really missed out on interpolation when setting up the escaping. 😅

**Comment by straight-shoota** [comment]
Would you mind also adding a spec for `BakedFileSystem::Loader::Encoder` like this?
```crystal
describe "Loader::Encoder" do
  it "escapes string interpolation" do
    String.build do |str|
      Encoder.new(str) << "foo\#{bar}"
    end.should eq "foo\\\#{bar} "
  end
end
```

**Comment by straight-shoota** [comment]
@jreinert Are you still up for this?

File: spec/storage/string_interpolation/test.txt (hunk 1 above)
Comment (repomaa) [reply spec/storage/string_interpolation/test.txt:1]: i can't modify the file without messing up the test. I named the folder to give hints on what it's for. Further explanation in the test.

**Comment by straight-shoota** [comment]
This was superseeded by #15 
Thanks @jreinert for setting the basis for that!

**Pull request closed without merging**